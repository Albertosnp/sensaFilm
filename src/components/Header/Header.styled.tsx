import styled from '@emotion/styled';
import breakpoints from '@/assets/styles/breakpoints';
import { rem } from '@/assets/styles/_mixins';

const HeaderStyled = styled.header`
  color: #ffffff;
  padding: 5% 5% 2% 5%;

  .header {
    &__title {
      font-size: ${rem(42)};
      text-shadow: black 0.1em 0.1em 0.2em;
    }
  }
  @media (min-width: ${rem(breakpoints.tablet)}) {
  }
  @media (min-width: ${rem(breakpoints.tablet_wide)}) {
  }
  @media (min-width: ${rem(breakpoints.desktop)}) {
  }
`;
export default HeaderStyled;
