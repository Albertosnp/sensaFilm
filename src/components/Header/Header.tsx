import HeaderStyled from './Header.styled';

export const Header = () => {
  return (
    <HeaderStyled className="header">
      <h1 className="header__title">SensaFILM</h1>
    </HeaderStyled>
  );
};
