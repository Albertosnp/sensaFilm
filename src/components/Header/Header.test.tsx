import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';

import { Header } from "./Header";

describe('Header', () => { 
    const defaultProps = {};
    const getComponent = (props = defaultProps) => <Header {...props} />;

    const setup = (props = defaultProps) => render(getComponent(props));
  
    it('Should render without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(getComponent(), div);
      ReactDOM.unmountComponentAtNode(div);
    });

    it('Should match snapshot', () => {
      const { container } = setup();
      expect(container).toMatchSnapshot();
    });
 })