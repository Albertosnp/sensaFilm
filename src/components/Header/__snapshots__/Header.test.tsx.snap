// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header > Should match snapshot 1`] = `
<div>
  <header
    class="header css-segxi8"
  >
    <h1
      class="header__title"
    >
      SensaFILM
    </h1>
  </header>
</div>
`;
